File: v2/client/src/types/foodItem.ts
import { exampleNutrient, exampleNutrientCarbs, exampleNutrientEnergy, exampleNutrientFat, exampleNutrientProtein, Nutrient } from "./nutrient";
import { UnitConversion } from "./unitConversion";

export type FoodItem = {
    id: number;
    name: string;
    verified: boolean;
    source: string;
    authorId: string;
    description: string;
    hasNutrition: boolean;
    nutrients: Nutrient[];
    unitConversions: UnitConversion[];
}

export const exampleFoodItem: FoodItem = {
    id: 1,
    name: 'Example Ingredient',
    verified: false,
    source: 'Example Source',
    authorId: 'author123123ID',
    description: 'This is the description of the example food item',
    hasNutrition: true,
    nutrients: [exampleNutrient, exampleNutrientEnergy, exampleNutrientProtein, exampleNutrientCarbs, exampleNutrientFat],
    unitConversions: [
        // Conversion from grams to other units
        {
            unit: 'g',
            multiplier: 1
        },
        {
            unit: 'kg',
            multiplier: 1000
        },
        {
            unit: 'lb',
            multiplier: 220.462
        }
    ]
};

File: v2/server/Core/Model/FoodItem.cs
namespace server.Core.Model;

public class FoodItem
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = "";
    public string Name { get; set; } = "";
    public bool Verified { get; set; } = false;

    /// <summary>
    /// Source of the food item. I.e. User, Frida, USDA, etc.
    /// </summary>
    public Source Source { get; set; }
    public string? AuthorId { get; set; }
    public string? Description { get; set; }
    public bool HasNutrition { get { return Nutrients.Count > 0; } }
    
    /// <summary>
    /// Nutrients are per gram of the food item.
    /// </summary>
    public List<Nutrient> Nutrients { get; set; } = [];

    /// <summary>
    /// Unit conversions are from grams to a different unit.
    /// </summary>
    public List<UnitConversion> UnitConversions { get; set; } = [];
}